error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/OptionalOrElseGet.java

```diff
@@ -36,12 +34,12 @@
  * it does, the suggested fix changes the program's semantics. Such fragile code must instead be
  * refactored such that the side-effectful code does not appear accidental.
  */
-// XXX: Consider also implementing the inverse, in which `.orElseGet(() -> someConstant)` is
-// flagged.
-// XXX: Once the `MethodReferenceUsageCheck` becomes generally usable, consider leaving the method
-// reference cleanup to that check, and express the remainder of the logic in this class using a
-// Refaster template, i.c.w. a `@Matches` constraint that implements the `requiresComputation`
-// logic.
+// XXX: This rule may introduce a compilation error: the `value` expression may reference a
+// non-effectively final variable, which is not allowed in the replacement lambda expression.
+// Review whether a `@Matcher` can be used to avoid this.
```

Comment:
This observation was only moved from the Refaster rule to here. Fixing it is out of scope.